tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/XPathAPI.java

```diff
@@ -2,26 +2,127 @@
         summary = "Fetch XPath string from hex id",
         description = "Looks up a specific hex ID.")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "404",
+                description = "XPath not found",
+                content =  @Content(mediaType = "application/json",
+                schema = @Schema(implementation = STError.class))),
+            @APIResponse(
+                responseCode = "200",
+                description = "Details about a given XPath",
+                content =  @Content(mediaType = "application/json",
+                schema = @Schema(implementation = XPathInfo.class))) })
     public Response getByHex(
-        @Parameter(required = true, example = "1234abcd") String hexId) {
+        @Parameter(required = true, example = "1234abcd", schema = @Schema(type = SchemaType.STRING))
+        @PathParam("hexId") String hexId) {
 
         // the actual implementation
-        return Response.ok(CookieSession.sm.xpt.getByStringID(hexId)).build();
+        XPathTable xpt = getXPathTable();
+        final String xpath = xpt.getByStringID(hexId);
+        if(xpath == null) {
+            return Response.status(Response.Status.NOT_FOUND)
+                .entity(new STError("XPath " + hexId + " not found"))
+                .build();
+        }
+        return Response.ok(new XPathInfo()
+            .setHexId(hexId)
+            .setDecimalId(xpt.getXpathIdFromStringId(hexId))
+            .setXpath(xpath)).build();
     }
 
+    @GET @Path("/dec/{decId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "Fetch XPath string from decimal id",
+        description = "Looks up a specific decimal ID.")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "404",
+                description = "XPath not found",
+                content =  @Content(mediaType = "application/json",
+                schema = @Schema(implementation = STError.class))),
+            @APIResponse(
+                responseCode = "200",
+                description = "Details about a given XPath",
+                content =  @Content(mediaType = "application/json",
+                schema = @Schema(implementation = XPathInfo.class))) })
+    public Response getByDecimal(
+        @Parameter(required = true, example = "5678", schema = @Schema(type = SchemaType.INTEGER))
+        @PathParam("decId") int decId) {
+
+        // the actual implementation
+        XPathTable xpt = getXPathTable();
+        final String xpath = xpt.getById(decId);
+        if(xpath == null) {
+            return Response.status(Response.Status.NOT_FOUND)
+                .entity(new STError("XPath #" + decId + " not found"))
+                .build();
+        }
+        return Response.ok(new XPathInfo()
+            .setDecimalId(decId)
+            .setHexId(xpt.getStringIDString(decId))
+            .setXpath(xpath)).build();
+    }
```

Comment:
This is awesome! A big difference from r.put -- maybe we'll still have some use for r.put -- anyway it's going to take a while to transition